src/layout/msWorkspace/msWindow.js

```diff
@@ -738,6 +741,7 @@ var MsWindow = GObject.registerClass(
         }
 
         _onDestroy() {
+            if (this.destroyId) this.disconnect(this.destroyId);
```

Comment:
This shouldn't be here because you can't disconnect a signal on a destroyed Actor and in the _onDestroy function it's already destroyed

Reply:
Even if _Destroy is called from another function?
https://github.com/material-shell/material-shell/blob/21b48f8c49efe4623391d750bafbcbe95b29cd22/src/layout/msWorkspace/msWindow.js#L681

Reply:
Maybe you didn't made all the possible tests?
I agree this part of code a is bit dirty.
We just need to consider the following deletion cases:
- The user close the taskitem tab and use our own killing feature. This way we destroy the metaWindows binded and then destroy the MsWindow. In this case we don't want to disconnect signal either to the destroyed metaWindows or to ourselves.
- The user close the metaWindow from another external way, metaWindow crash or closed from the overview or close button. In this case the metaWindow is already destroyed when entering in the Kill method but we need to destroy the MsWindow. So here we don't want to disconnect on metaWindow and neither on MsWindow being destroyed.
- And the last case it's when we destroy the MsWindow but without destroying the metaWindow. When we disable the extension for exemple. In this case we want to disconnect all metaWindow signals and then destroy.

And we need to be aware that when shutting down the session the MetaWindow and the MsWindow will be destroyed and i'm not sure the order is stable